packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,273 @@
+  const alertTypeClassName = getAlertClassName(type);
+  const narrowAlertClassNames = classNames([
+    'terra-Alert',
+    'terra-Alert--narrow',
+    alertTypeClassName,
+    attributes.className,
+  ]);
+  const wideAlertClassNames = classNames([
+    'terra-Alert',
+    'terra-Alert--wide',
+    alertTypeClassName,
+    attributes.className,
+  ]);
+
+  let actionsSection = '';
+  let dismissButton = '';
+  const outerDivStyle = {};
+  let alertSectionClassName = 'terra-Alert-section';
+  let alertActionsClassName = 'terra-Alert-actions';
+
+  if (type === AlertTypes.CUSTOM) {
+    // For custom alert, there is no color assigned to the box-shadow style since it is to be specified
+    // in the customStatusColor prop.  The box-shadow style is defined in CSS in order to get the
+    // bidirectionality via the mixin.  As per the W3C spec if the box-shadow does not have the color
+    // defined, it will use the prevailing color style, so setting that here. But then we need to set
+    // the color style for the alert content so that it doesn't pick up the custom status color. We
+    // will allow the icon to pick up the color style so that Terra icons will match the color of the
+    // status bar.
+    outerDivStyle.color = customStatusColor;
+    alertSectionClassName += ' terra-Alert-section--custom';
+    alertActionsClassName += ' terra-Alert-actions--custom';
+  }
+
+  if (onDismiss) {
+    dismissButton = (<Button text={intl.formatMessage({ id: 'Terra.alert.dismiss' })} size="medium" variant="secondary" onClick={onDismiss} />);
+  }
+  if (onDismiss || alertAction) {
+    actionsSection = (
+      <div className={alertActionsClassName}>
+        {alertAction || ''}
+        {dismissButton}
+      </div>
+    );
+  }
+
+  const alertMessageContent = (
+    <div className={alertSectionClassName}>
+      <strong className="terra-Alert-title">{title || defaultTitle}</strong>
+      <div className="terra-Alert-content">
+        {children}
+      </div>
+    </div>
+  );
+
+  return (
+    <ResponsiveElement
+      responsiveTo="parent"
+      defaultElement={
+        <div {...attributes} className={narrowAlertClassNames} style={outerDivStyle} >
+          <div className="terra-Alert-body terra-Alert-body-narrow">
+            <div className="terra-Alert-icon">{getAlertIcon(type, customIcon)}</div>
+            {alertMessageContent}
+          </div>
+          {actionsSection}
+        </div>
+      }
+      tiny={
+        <div {...attributes} className={wideAlertClassNames} style={outerDivStyle} >
+          <div className="terra-Alert-body terra-Alert-body-wide">
+            <div className="terra-Alert-icon">{getAlertIcon(type, customIcon)}</div>
+            {alertMessageContent}
+          </div>
+          {actionsSection}
+        </div>
+      }
+    />
+
+  );
+};
+
+Alert.propTypes = propTypes;
+Alert.defaultProps = defaultProps;
+Alert.contextTypes = contextTypes;
+Alert.Types = AlertTypes;
```

Comment:
nit: I'd like to change this to `Alert.Opts.Types = AlertTypes`. The thinking behind this is that if we want to add any other constants for prop values, we can add those to the Opts object. This is a pattern we could use across all components.